Makefile

```diff
@@ -2,19 +2,22 @@ SHELL := /bin/bash
 
 OSX_MIN_VERSION := 10.9
 
-DOCKER := $(shell which docker 2> /dev/null)
-GO := $(shell which go 2> /dev/null)
-NODE := $(shell which node 2> /dev/null)
-NPM := $(shell which npm 2> /dev/null)
-S3CMD := $(shell which s3cmd 2> /dev/null)
-WGET := $(shell which wget 2> /dev/null)
-RUBY := $(shell which ruby 2> /dev/null)
+get-command = $(shell which="$$(which $(1) 2> /dev/null)" && if [[ ! -z "$$which" ]]; then printf %q "$$which"; fi)
```

Comment:
My Windows setup was on a path that contained spaces and this was interfering with the $(NPM) target. I ended up removing the $(NPM) target in favour of a descriptive message but I also fixed the problems with spaces on file names by passing them through `printf %q`.